Libraries/Image/RCTNetworkImageView.m

```diff
@@ -95,13 +135,16 @@ - (void)setImageURL:(NSURL *)imageURL resetToDefaultImageWhileLoading:(BOOL)rese
             self.layer.minificationFilter = kCAFilterLinear;
             self.layer.magnificationFilter = kCAFilterLinear;
             [self.layer addAnimation:animation forKey:@"contents"];
+            loadEndHandler();
           });
         } else if (error) {
-          RCTLogWarn(@"Unable to download image data. Error: %@", error);
+          errorHandler([error description]);
         }
       }];
     } else {
-      _downloadToken = [_imageDownloader downloadImageForURL:imageURL size:self.bounds.size scale:RCTScreenScale() resizeMode:self.contentMode backgroundColor:self.backgroundColor block:^(UIImage *image, NSError *error) {
+      _downloadToken = [_imageDownloader downloadImageForURL:imageURL size:self.bounds.size scale:RCTScreenScale()
+                                                  resizeMode:self.contentMode backgroundColor:self.backgroundColor
+                                                  progressBlock:progressHandler block:^(UIImage *image, NSError *error) {
```

Comment:
Thanks for comments, I renamed events and tryed to not fire the progress event if it's not defined.

Tested with something like this: 

```javascript
class TestNetworkImage extends Component {
  componentDidMount() {
      setTimeout(() => {
        this.setState({
          onProgress: this.onProgress.bind(this)
        })
      }, 1000);
    }

  render() {
    return <Image ref="image"
      source={{uri: this.props.source}}
      style={this.props.style}
      onLoadProgress={this.state.onProgress}>
      {this.props.loader}
      {this.props.children}
    </Image>
  }
}
```
`NSLog` shows that at first second ImageView doesn't send an event to js in that case, then it does.